aws/logs_monitoring/template.yaml

```diff
@@ -718,51 +546,47 @@ Resources:
-        Fn::If:
-          - SetPermissionsBoundary
-          - Ref: PermissionsBoundaryArn
-          - Ref: AWS::NoValue
+      PermissionsBoundary: !If
+        - SetPermissionsBoundary
+        - !Ref PermissionsBoundaryArn
+        - !Ref AWS::NoValue
       Policies:
         - PolicyName: ForwarderRolePolicy0
           PolicyDocument:
             Version: "2012-10-17"
             Statement:
               # Access the s3 bucket that is used by the forwarder as a datastore
-              - Fn::If:
-                  - SetForwarderBucket
-                  - Action:
-                      - s3:GetObject
-                      - s3:PutObject
-                      - s3:DeleteObject
-                    Resource:
-                      - Fn::If:
-                          - CreateS3Bucket
-                          - Fn::Join:
-                              - "/"
-                              - - Fn::GetAtt: ForwarderBucket.Arn
-                                - "*"
-                          - Fn::Sub: "arn:aws:s3:::${DdForwarderExistingBucketName}/*"
-                    Effect: Allow
-                  - Ref: AWS::NoValue
+              - !If
+                - SetForwarderBucket
+                - Action:
+                    - s3:GetObject
+                    - s3:PutObject
+                    - s3:DeleteObject
+                  Resource:
+                    - !If
+                      - CreateS3Bucket
+                      - !Sub "${ForwarderBucket.Arn}/*"
+                      - !Sub "arn:aws:s3:::${DdForwarderExistingBucketName}/*"
+                  Effect: Allow
+                - !Ref AWS::NoValue
               # Get the actual log content from the s3 bucket based on the received s3 event.
               # Use PermissionsBoundaryArn to limit (allow/deny) access if needed.
-              - Fn::If:
-                  - SetForwarderBucket
-                  - Action:
-                      - s3:ListBucket
-                    Resource:
-                      - Fn::If:
-                          - CreateS3Bucket
-                          - Fn::GetAtt: ForwarderBucket.Arn
-                          - Fn::Sub: "arn:aws:s3:::${DdForwarderExistingBucketName}"
-                    Condition:
-                      StringLike:
-                        s3:prefix:
-                          - "retry/*"
-                          - "log-group-cache/*"
-                    Effect: Allow
-                  - Ref: AWS::NoValue
+              - !If
+                - SetForwarderBucket
+                - Action:
+                    - s3:ListBucket
+                  Resource:
+                    - !If
+                      - CreateS3Bucket
+                      - !GetAtt ForwarderBucket.Arn
+                      - !Sub "arn:aws:s3:::${DdForwarderExistingBucketName}"
+                  Condition:
+                    StringLike:
+                      s3:prefix:
+                        - "failed_events/*"
```

Comment:
is change `retry` -> `failed_events` intended?

Reply:
good catch, it will be in the next PR